validation/linux_readonly_paths.go

```diff
@@ -28,13 +45,130 @@ func main() {
 		}
 		defer os.Remove(tmpfile.Name())
 
-		testFile := filepath.Join(path, "readonly-file")
-
 		// runtimetest cannot check the readability of empty files, so
 		// write something.
+		testSubSubFile := filepath.Join(path, readonlyFileSubSub)
+		if err := ioutil.WriteFile(testSubSubFile, []byte("immutable"), 0777); err != nil {
+			return err
+		}
+
+		testSubFile := filepath.Join(path, readonlyFileSub)
+		if err := ioutil.WriteFile(testSubFile, []byte("immutable"), 0777); err != nil {
+			return err
+		}
+
+		testFile := filepath.Join(path, readonlyFile)
 		return ioutil.WriteFile(testFile, []byte("immutable"), 0777)
 	})
+	return err
+}
+
+func checkReadonlyRelPaths() error {
+	g, err := util.GetDefaultGenerator()
+	if err != nil {
+		return err
+	}
+
+	// Deliberately set a relative path to be read-only, and expect an error
+	readonlyRelPath := "/etc/../readonly-relpath"
```

Comment:
Is it a relative path? I think it equal to `/readonly-relpath`. It's a absolute path.

Reply:
@q384566678 Fixed the relative path.
Also fixed the same one in https://github.com/opencontainers/runtime-tools/pull/643.

Reply:
Why not  use `readonlyRelPath := "readonly-relpath"` ?

Reply:
@q384566678 Done. Also done in #643.